packages/core-dialog/core-dialog.js

```diff
@@ -25,23 +25,37 @@ export default function dialog (dialogs, open) {
     dialog.setAttribute('aria-modal', true)
     dialog.setAttribute('aria-label', options.label || dialog.getAttribute('aria-label'))
     hasBackdrop || dialog.insertAdjacentElement('afterend', document.createElement('backdrop'))
+    options.strict && dialog.setAttribute('data-dialog-strict', options.strict)
 
     setOpen(dialog, options.open)
     return dialog
   })
 }
 
+// Focus can move outside the dialog when in "strict"-mode, therefore
+// we need to ensure that if that happens we will refocus on the dialog.
+addEvent(UUID, 'focus', (event) => {
+  let hasDialog = false
+  for (let el = event.target; el; el = el.parentElement) {
+    if (el.hasAttribute(UUID)) return (hasDialog = true)
+  }
+
+  const dialog = getTopLevelDialog()
+  if (!hasDialog && dialog) setFocus(dialog)
+}, true)
+
 addEvent(UUID, 'click', (event) => {
   // This functions handles if the user clicked on a open or close button.
   // We need to do this loop in order for the close button to know which
   // dialog it should close.
   for (let el = event.target, isClose; el; el = el.parentElement) {
     const action = el.getAttribute(ATTR)
     const prev = el.previousElementSibling
+    const isStrict = prev && prev.hasAttribute(UUID) && Boolean(prev.getAttribute('data-dialog-strict'))
     isClose = isClose || action === 'close'
```

Comment:
And change this to
```
change to:
```
isClose = isClose || (prev && prev.getAttribute(UUID) === 'false' && (el = prev)) || action === 'close'
```